File: 0041-first-missing-positive/0041-first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {

     unordered_map<int , int> mp;
         int ans;
     for(int i = 0; i < nums.size(); i++){
        if(nums[i] > 0)
        mp[nums[i]]++;
     }   
 
     for(int i = 1; i <= nums.size(); i++){
        if(mp.find(i) == mp.end()){
            return  i;
        }
       
       
       
     }
     return nums.size()+1;
    }
};

File: 0041-first-missing-positive/0041-first-missing-positive.py
class Solution(object):
    def firstMissingPositive(self, nums):
        mp = {}
        

        for i in nums:
            if i > 0:
                mp[i] = mp.get(i,0) + 1

        size = len(nums) + 1

        for i in range(1 , size):
            if i not in mp:
                return i

        return len(nums) + 1

        